pkgs/http_client_conformance_tests/lib/src/redirect_tests.dart

```diff
@@ -2,60 +2,40 @@
 // for details. All rights reserved. Use of this source code is governed by a
 // BSD-style license that can be found in the LICENSE file.
 
-import 'dart:async';
-import 'dart:io';
-
 import 'package:http/http.dart';
+import 'package:stream_channel/stream_channel.dart';
 import 'package:test/test.dart';
 
 /// Tests that the [Client] correctly implements HTTP redirect logic.
 void testRedirect(Client client) async {
   group('redirects', () {
-    late HttpServer server;
+    late String host;
+    late StreamChannel<Object?> httpServerChannel;
+
     setUp(() async {
-      //        URI |  Redirects TO
-      // ===========|==============
-      // ".../loop" |    ".../loop"
-      //   ".../10" |       ".../9"
-      //    ".../9" |       ".../8"
-      //        ... |           ...
-      //    ".../1" |           "/"
-      //        "/" |  <no redirect>
-      server = (await HttpServer.bind('localhost', 0))
-        ..listen((request) async {
-          if (request.requestedUri.pathSegments.isEmpty) {
-            unawaited(request.response.close());
-          } else if (request.requestedUri.pathSegments.last == 'loop') {
-            unawaited(request.response
-                .redirect(Uri.http('localhost:${server.port}', '/loop')));
-          } else {
-            final n = int.parse(request.requestedUri.pathSegments.last);
-            final nextPath = n - 1 == 0 ? '' : '${n - 1}';
-            unawaited(request.response
-                .redirect(Uri.http('localhost:${server.port}', '/$nextPath')));
-          }
-        });
+      httpServerChannel = spawnHybridUri('../lib/src/redirect_server.dart');
+      host = 'localhost:${await httpServerChannel.stream.first as int}';
     });
-    tearDown(() => server.close);
+    tearDown(() => httpServerChannel.sink.add(null));
 
     test('disallow redirect', () async {
-      final request = Request('GET', Uri.http('localhost:${server.port}', '/1'))
+      final request = Request('GET', Uri.http(host, '/1'))
         ..followRedirects = false;
       final response = await client.send(request);
       expect(response.statusCode, 302);
       expect(response.isRedirect, true);
-    });
+    }, onPlatform: {'browser': const Skip('not passing')});
```

Comment:
Should we consider a boolean argument expressing this limitation? The browser client doesn't honor `followRedirect` or `maxRedirects` at all, so we could consider skipping any test which relies on those with `canControlRedirects: false`?

Reply:
I'm not sure. If we are trying to catch issues that **could** be fixed then we want the skip here - because you could implement BrowserClient in a way that would fix this.

Reply:
I'm convinced - fixed!